### PR TITLE
React 18 in peer-dependencies and use new API

### LINKED ISSUE #1102: React 18 Support
I apologize if this is the wrong place for the issue, but I'm trying to upgrade to React 18 and running into issues with this library. I can see in package.json it's looking for React 16 or 17 so I'm curious if there are any plans to support 18. Thanks

### PR DESCRIPTION
**Pull request by derkoe**
Closes #1102

### CODE REVIEW

**Comment by AlexcastroDev** [comment]
Thanks @derkoe ❤️

**Comment by fahadlabs** [comment]
Waiting for this change.

**Comment by BranChembah123** [comment]
I hope this will also solve #1110

**Comment by BrandonCopley** [comment]
How can I help get this into production?

**Comment by tetarchus** [comment]
@karldanninger @itsmichaeldiego - Are you able to merge and release this PR to allow upgrading to React 18?

**Review by geangu — APPROVED** [review]

**Review by bnussman — APPROVED** [review]

File: example/src/App.test.js [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
 import ReactDOM from 'react-dom'
 import App from './App'
+import { createRoot } from 'react-dom/client';
+
 
 it('renders without crashing', () => {
   const div = document.createElement('div')
-  ReactDOM.render(<App />, div)
+  const root = createRoot(div);
+  root.render(<App />)
```
Comment (stephencweiss) [inline example/src/App.test.js:10]: Does this actually maintain backward compatibility? Specifically - does React 16/17 _have_ a `createRoot`? Or does this just fix it for React 18?

File: example/src/index.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import './index.css'
 
 import React from 'react'
-import ReactDOM from 'react-dom'
 import App from './App'
 
-ReactDOM.render(<App />, document.getElementById('root'))
+const root = createRoot(document.getElementById('root'))
```
Comment (stephencweiss) [inline example/src/index.js:6]: I suppose this is actually where I meant to ask the question about backwards compatibility.

File: example/src/App.test.js (hunk 1 above)
Comment (derkoe) [reply example/src/App.test.js:10]: This is just an example. In your app you have to use the API that works with your React version

**Comment by joaogarin** [comment]
Would be lovely to get this released and unblock people using the library to move to React 18 =)

**Comment by agencekonig** [comment]
Please release it 👯

**Comment by asdutoit** [comment]
??

**Comment by DianaMachadoMoura** [comment]
Could you please release it? 🙏

**Comment by karldanninger** [comment]
Sorry this took so long... We need to get better at maintain this package.
I've merged this and [set a release 2.2](https://github.com/google-map-react/google-map-react/releases/tag/2.2)

Let me know if this is enough for npm to serve this latest version.

**Comment by jacobgad** [comment]
i don't believe this is enough for npm to serve the latest version. As you can see here:
https://www.npmjs.com/package/google-map-react

it is still serving 2.1.10

**Comment by derkoe** [comment]
@karldanninger somebody has to publish the package manually since there are no CI jobs setup (the last version was published by @itsmichaeldiego)

**Comment by itsmichaeldiego** [comment]
Done @karldanninger @derkoe 

Let me know!

**Comment by asdutoit** [comment]
Many thanks!

**Comment by itsmichaeldiego** [comment]
@derkoe @karldanninger @dryan-FF Seems that there's an issue still https://github.com/google-map-react/google-map-react/issues/1129#issuecomment-1239111262, could you guys check it?

**Pull request merged**